src/index.tsx

```diff
@@ -594,6 +594,29 @@ class UserJourneyTracker {
         return urlAttribution;
     }
 
+    /**
+     * Check if the new event's attribution is the same as the last stored event's attribution
+     * @param newAttribution The attribution data of the new event
+     * @returns True if the event is duplicated (same attribution), false otherwise
+     */
+    private isEventDuplicated(newAttribution: AttributionData): boolean {
+        if (this.events.length === 0) return false;
+
+        const lastEvent = this.events[this.events.length - 1];
+        const lastAttribution = lastEvent.attribution;
+
+        // Compare all keys in newAttribution and lastAttribution
+        const keys = new Set([...Object.keys(newAttribution), ...Object.keys(lastAttribution)]);
```

Comment:
you need to ignore some of the keys as discussed like the timestamp, ...

Reply:
added for timestamp and landing_page